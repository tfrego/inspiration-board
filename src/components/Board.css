.board {
  display: flex;
  flex-wrap: wrap;
  background-color: gray;
  padding: 1rem 1rem;
}

.validation-errors-display {
  text-align: center;
  margin-bottom: 2em;
}

.validation-errors-display__list {
  list-style-type: none;
  padding: 0;
}
