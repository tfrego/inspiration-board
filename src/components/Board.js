import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';
// import CARD_DATA from '../data/card-data.json';

class Board extends Component {
  constructor(props) {
    super(props);

    this.state = {
      cards: [],
      boardUrl: `${this.props.boardUrl}${this.props.boardName}/cards`,
    };
  }

  componentDidMount() {

    axios.get(this.state.boardUrl)
      .then((response) => {
        const cards = response.data.map((card) => {
          const newCard = {
            ...card.card,
          }
          return newCard;
        })
        this.setState({cards: cards});
        console.log(this.state);
      })
  }

  deleteCard = (cardId) => {
    console.log(cardId);
    axios.delete(`${this.props.cardUrl}${cardId}`)
      .then( (response) => {
        console.log(response);
        const cards = this.state.cards;

        const card = cards.find( (card) => {
          return card.id === cardId;
        })

        cards.splice( cards.indexOf(card), 1 );

        this.setState({cards: cards})
      })
      .catch( (error) => {
        this.setState({
          errorMessage: `Failure ${error.message}`,
        })
      })
  }

  addCard = (newCard) => {
    console.log(newCard);
    const apiPayLoad = {
      text: newCard.text,
      emoji: newCard.emoji,
    }

    axios.post(this.state.boardUrl, apiPayLoad)
      .then( (response) => {
        console.log(response);
        const myNewCard = response.data;
        console.log(myNewCard);

        const cards = this.state.cards;
        cards.push(myNewCard.card);

        this.setState({cards: cards});
        console.log(this.state)
      })
      .catch( (error) => {
        this.setState({
          errorMessage: `Failure ${error.message}`,
        })
      })
  }

  render() {
    const cardCollection = this.state.cards.map((card, i) => {
      return <Card
              key={card.id}
              {...card}
              deleteCardCallback={this.deleteCard}
              />
    })
    return (
      <section>
        <NewCardForm addCardCallback={this.addCard} />
        <div className="board">
          {cardCollection}
        </div>
      </section>
    )
  }

}

Board.propTypes = {
  boardUrl: PropTypes.string.isRequired,
  cardUrl: PropTypes.string,
  boardName: PropTypes.string.isRequired,
};

export default Board;
