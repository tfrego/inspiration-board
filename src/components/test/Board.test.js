import React from 'react';
import Board from '../Board';
import { shallow } from 'enzyme';

describe('board', () => {
  it('will match the Board Snapshot', () => {
    const wrapper = shallow( <Board boardUrl="board_url" cardUrl="card_url" boardName="test" />);

    expect(wrapper).toMatchSnapshot();
  });
});
