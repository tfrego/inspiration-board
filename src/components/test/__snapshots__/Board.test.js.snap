// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`board will match the Board Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board
    boardName="test"
    boardUrl="board_url"
    cardUrl="card_url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NewCardForm
          addCardCallback={[Function]}
        />,
        <div
          className="board"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addCardCallback": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "board",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NewCardForm
            addCardCallback={[Function]}
          />,
          <div
            className="board"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addCardCallback": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "board",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
